src/pages/Login.jsx

```diff
@@ -1,75 +1,205 @@
 import React from 'react';
-import logo from "../assets/images/logo.png";
-import Gradient from "../assets/images/mesh-gradient.png";
-import "../assets/styles/Login.css"
-import google from "../assets/svg/google.svg"
+import { useState,useRef,useEffect } from 'react';
+import "../assets/styles/Login.css";
 import Navbar from "../components/Navbar.jsx";
-import Footer from "../components/Footer.jsx";
+import { Link } from 'react-router-dom';
+import SignUp from './SignUp.jsx';
+
+const USER_REGEX = /^[A-Za-z][A-Za-z0-9_]{3,29}$/; // 4-30 characters
+const PWD_REGEX = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%]).{8,24}$/;
```

Comment:
Improve this regex for password

Reply:
Can you explain what changes can be done to improve it?